### PR TITLE
Add secure headers for all resources.

### PR DESCRIPTION
**Pull request by ewelinajozwik**
Apply relevant headers to all responses, following [OWASP Secure Headers Project guides](https://www.owasp.org/index.php/OWASP_Secure_Headers_Project#tab=Headers).

Relates to [VRE-821](https://thehyve.atlassian.net/browse/VRE-821)

Remaining headers that were not added:

- `Strict-Transport-Security` - already added by default
- `Content-Security-Policy` - making strict policies here breaks the user interface, might be specific per installation

Headers that do not seem to bring any benefit (since we already have Pluto middle-layer with zuul taking care of filtering and routing rules):

- `Referrer-Policy`
- `Cross-Origin-Embedder-Policy`
- `Cross-Origin-Opener-Policy`
- `Cross-Origin-Resource-Policy`

### CODE REVIEW

**Comment by gijskant** [comment]
For comparison, for the Palga portal we use:
- `Content-Security-Policy-Report-Only: default-src 'self' 'unsafe-inline'; img-src 'self' data:; font-src 'self' data:`
- `X-XSS-Protection: 1; mode=block`

**Comment by ewelinajozwik** [comment]
> `X-XSS-Protection: 1; mode=block`

The `X-XSS-Protection` is deprecated and the OWASP guide suggests to set it to 0: https://owasp.org/www-project-secure-headers/#x-xss-protection

The `Content-Security-Policy-Report-Only` was not on the list, only the `Content-Security-Policy`. I can check what is the difference. I tried a similar values for `Content-Security-Policy` and the UI was not showing up (at least locally), but I can still try to tune this a bit. I'm just afraid it depends on the installation and may break things.

**Comment by ewelinajozwik** [comment]
I added `Content-Security-Policy`. From what I checked `Content-Security-Policy-Report-Only` only prints violation reports to the console and delivers them to a violation endpoint if the report-to and report-uri directives are used (we don't have such an endpoint), but it does not block violations.

I did not add a policy for image-src, because it was indeed blocking images in the markdown field.

**Comment by gijskant** [comment]
> I did not add a policy for image-src, because it was indeed blocking images in the markdown field.

But then it uses the policy in `default-src` as fallback (which means rejecting in this case).

File: projects/pluto/src/main/java/nl/fairspace/pluto/auth/filters/SecureResponseHeaderAppenderFilter.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    public void doFilter(ServletRequest request, ServletResponse response,
+                         FilterChain chain) throws IOException, ServletException {
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+        httpServletResponse.setHeader("X-Frame-Options", "DENY");
+        httpServletResponse.setHeader("X-Content-Type-Options", "nosniff");
+        httpServletResponse.setHeader("X-XSS-Protection", "0");
+        httpServletResponse.setHeader("X-Permitted-Cross-Domain-Policies", "none");
+        httpServletResponse.setHeader(
+                "Content-Security-Policy",
+                "default-src 'self' 'unsafe-inline'; script-src 'self' 'unsafe-inline'; font-src 'self'; style-src: 'self'"
```
Comment (gijskant) [inline projects/pluto/src/main/java/nl/fairspace/pluto/auth/filters/SecureResponseHeaderAppenderFilter.java]: - `script-src` and `style-src` can inherit from `default-src`
- Allow including images from all https addresses
```suggestion
                "default-src 'self' 'unsafe-inline'; font-src 'self'; img-src 'self' 'unsafe-inline' https://*"
```

Comment (ewelinajozwik) [reply projects/pluto/src/main/java/nl/fairspace/pluto/auth/filters/SecureResponseHeaderAppenderFilter.java]: Thanks for the suggestion! I applied it.

**Review by gijskant — APPROVED** [review]
Advised headers applied. Application is still working as expected. External images, referenced in Markdown fields are allowed, if https is used, otherwise images are blocked (as they should).
(When we add a description of the Markdown format somewhere, this could be mentioned.)

**Pull request merged**